src/ray/core_worker/core_worker.cc

```diff
@@ -2623,6 +2625,15 @@ Status CoreWorker::ExecuteTask(
 
   if (!options_.is_local_mode) {
     task_counter_.MoveRunningToFinished(func_name, task_spec.IsRetry());
+    task_manager_->RecordTaskStatusEvent(
+        task_spec.AttemptNumber(),
+        task_spec,
+        /* We will record the terminal state on the submitter worker */
+        rpc::TaskStatus::NIL,
+        /* include_task_info */ false,
+        /* node_id */ absl::nullopt,
+        /* worker_id */ absl::nullopt,
+        task_log_info);
```

Comment:
Hmm why do we need to add the call here, could we record it on the owner side? That would save extra ask status messages.

Reply:
You mean the c++ CoreWorker? Yeah, that's an alternative, I guess the major implementation blockers I ran into was: 
- since we have the core worker id in the worker filename, we only know the exact file handle after CoreWorker is initialized so we couldn't pass the file handle as args to the owner. 


^ But I guess I could pass that info to the owner from `default_worker.py`

Reply:
What I meant was, why not return the log info as part of the task response, so that this info can be logged in the same RecordTaskStatusEvent call of moving the task state to FINISHED? This avoids an extra RPC to GCS.

Reply:
Ah I see. Sorry, misunderstood it. 

So the extra `RecordTaskStatusEvent` will simply add this into the local buffer on the coreworker, and be batched as one of the events in a single RPC call to GCS. So this should not affect the number of RPC calls since we are already pushing events from executor to GCS for RUNNING events. 

This might have one additional event in the batch: https://github.com/ray-project/ray/blob/b2c5e63bf7df2b105dbdcaeb558c682c9103712f/src/ray/protobuf/gcs.proto#L240

But this will quickly be optimized as we merge [`TaskStateUpdate`](https://github.com/ray-project/ray/blob/b2c5e63bf7df2b105dbdcaeb558c682c9103712f/src/ray/protobuf/gcs.proto#L201) from the coreworker side in this [PR](https://github.com/ray-project/ray/pull/32574)

Reply:
Got it.